plugins/paho-mqtt/src/main/java/com/navercorp/pinpoint/plugin/paho/mqtt/interceptor/CommsCallbackConstructorInterceptor.java

```diff
@@ -0,0 +1,91 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.pinpoint.plugin.paho.mqtt.interceptor;
+
+import com.navercorp.pinpoint.bootstrap.context.MethodDescriptor;
+import com.navercorp.pinpoint.bootstrap.context.TraceContext;
+import com.navercorp.pinpoint.bootstrap.interceptor.AroundInterceptor;
+import com.navercorp.pinpoint.bootstrap.logging.PLogger;
+import com.navercorp.pinpoint.bootstrap.logging.PLoggerFactory;
+import com.navercorp.pinpoint.plugin.paho.mqtt.PahoMqttPluginConfig;
+import com.navercorp.pinpoint.plugin.paho.mqtt.accessor.BrokerUriFieldAccessor;
+import org.eclipse.paho.mqttv5.client.MqttClientInterface;
+
+import static com.navercorp.pinpoint.plugin.paho.mqtt.PahoMqttConstants.UNKNOWN;
+
+/**
+ * @author Younsung Hwang
+ */
+public class CommsCallbackConstructorInterceptor implements AroundInterceptor {
+
+    private final PLogger logger = PLoggerFactory.getLogger(this.getClass());
+    private final boolean isDebug = logger.isDebugEnabled();
+
+    private PahoMqttPluginConfig config;
+    private final TraceContext traceContext;
+    protected final MethodDescriptor descriptor;
+
+    public CommsCallbackConstructorInterceptor(TraceContext traceContext, MethodDescriptor descriptor, PahoMqttPluginConfig config) {
+        this.traceContext = traceContext;
+        this.descriptor = descriptor;
+        this.config = config;
+    }
+
+    @Override
+    public void before(Object target, Object[] args) {
+        if (isDebug) {
+            logger.beforeInterceptor(target, args);
+        }
+    }
+
+    @Override
+    public void after(Object target, Object[] args, Object result, Throwable throwable) {
+        if (isDebug) {
+            logger.afterInterceptor(target, args, result, throwable);
+        }
+
+        if (throwable != null) {
+            return;
+        }
+
+        if (!(target instanceof BrokerUriFieldAccessor)) {
+            return;
+        }
+
+        String brokerUri = extractBrokerUri(args);
+        if (brokerUri != null) {
+            ((BrokerUriFieldAccessor) target)._$PINPOINT$_setBrokerUri(brokerUri);
+            return;
+        }
+    }
+
+    private String extractBrokerUri(Object[] args) {
+        // MQTT Version 3 case
+        if(this.config.isEnableTracePahoMqttClientV3() && args[0] instanceof org.eclipse.paho.client.mqttv3.internal.ClientComms){
+            org.eclipse.paho.client.mqttv3.internal.ClientComms clientComms = (org.eclipse.paho.client.mqttv3.internal.ClientComms)args[0];
+            org.eclipse.paho.client.mqttv3.IMqttAsyncClient mqttAsyncClient = clientComms.getClient();
+            return mqttAsyncClient.getServerURI();
+        }
+        // MQTT Version 5 case
+        if(this.config.isEnableTracePahoMqttClientV5() && args[0] instanceof org.eclipse.paho.mqttv5.client.internal.ClientComms){
```

Comment:
@koo-taejin You're right. I missed the case.

I created two classes: CommsCallbackV3ConstructorInterceptor and CommsCallbackV5ConstructorInterceptor.

I was thinking of eliminating source duplication by making CommsCallbackConceptorInterceptor an abstract class.
But I didn't do it because I thought it would make it more complicated.